### DISCUSSION TITLE
Request: Improve the documentation

### DISCUSSION THREAD

**Discussion by ElectreAAS** [ocaml/dune#15153] [Ideas]
This issue is part of a series of comments, criticisms, and requests on dune I gathered during my last visit at Inria. If some of these seem trivial or incoherent at face value, I'd wager we should press the authors for more context.
Everything below is paraphrased from my notes.

## Desired Behavior

Rewrite the whole documentation, maybe from scratch, it's bad.
It works as a specification or reference manual, but not as documentation.
Useful, but impractical.

**Comment by ElectreAAS** [comment]
(This comment is my own)
The comment above was made to me in what I suspect wasn't entirely good faith. And I've heard in the dune team the common knowledge "let's not try to improve dune for people who will never use it, no matter how good".
Nevertheless, I believe there is a kernel of valuable insight here:
The current documentation over at readthedocs says we should use the [diataxis](https://diataxis.fr/) mindset, and in their terminology, there is a lot of "reference", but not so much "explanation", "how-to guides", and in my opinion, even less "tutorial".

This doesn't mean this is an easily actionable subject, but I've been encouraged to publish this issue for general information anyway.

**Comment by emillon** [comment]
> I've heard in the dune team the common knowledge "let's not try to improve dune for people who will never use it, no matter how good".

I feel like I should rephrase what I said - to use a business analogy, there's a balance between working on user aquisition and retention. This is what I meant with "I'm not trying to get new users at all costs".

As for this particular issue, yes dune's docs are not ideal. The current situation is in an intermediate state: some pages haven't been moved (yet) to a proper location.

> there is a lot of "reference", but not so much "explanation", "how-to guides", and in my opinion, even less "tutorial".

Absolutely! This is something I wasn't aware of before trying to untangle the existing docs. We can't fix what we can't see. I'm working with the ocaml.org team so that these docs are available from ocaml.org eventually.

> This doesn't mean this is an easily actionable subject, but I've been encouraged to publish this issue for general information anyway.

Yes. We already have many open "the docs are not clear" issues. What would be, in your opinion, a way to close this particular one?

**Comment by rgrinberg** [comment]
I turned this into a discussion because it is too open ended to be an issue. Feel free to pursue ideas on how to improve the docs and create more focused issues.
